projects/cli-files/go-cat/cmd/root.go

```diff
@@ -0,0 +1,107 @@
+package cmd
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"strings"
+)
+
+func Execute() {
+	// define a flag of type boolean with the specific properties (name, value, usage)
+	numberFlag := flag.Bool("n", false, "number all output lines")
+
+	// parse the command line flags
+	flag.Parse()
+
+	// if there is no argument given then error
+	if len(os.Args) < 2 {
+		fmt.Fprintf(os.Stderr, "no filename provided\n")
+		os.Exit(2)
+	}
+
+	// create a variable to store the file name
+	var filename string
+
+	if *numberFlag {
+		// if the -n flag is provided the filename will be the third value in the os.Args slice
+		filename = os.Args[2]
+	} else {
+		// if no flag is provided the filename will be the second value in the os.Args slice
+		filename = os.Args[1]
+	}
+
+	// check if the file exists
+	fileInfo, err := os.Stat(filename)
+
+	if fileInfo.IsDir() {
+		fmt.Fprintf(os.Stderr, "%s : Is a directory\n", filename)
+		os.Exit(1)
+	}
+
+	// if the file does not exist then error
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "no such file or directory: %s\n", filename)
```

Comment:
nit: you can remove directory from error message

Reply:
oh scratch that, you should check for errors before checking for`fileInfo.IsDir()`

Go is very uforgiving when it comes to panics, so always check for errors before working with the returned value